helper/predeployment/predeployment.go

```diff
@@ -0,0 +1,254 @@
+		splitValues := strings.Split(arraySubstring, ",")
+
+		for _, str := range splitValues {
+			ret = append(ret, extractValue(str))
+		}
+	}
+
+	return ret
+}
+
+// normalizeConstructorArguments cleans up the constructor arguments so that they
+// can be interpreted by the ABI encoder
+func normalizeConstructorArguments(constructorArgs []interface{}) []interface{} {
+	arguments := make([]interface{}, 0)
+
+	for _, arg := range constructorArgs {
+		argStr, _ := arg.(string)
+		arguments = append(arguments, extractValue(argStr))
+	}
+
+	return arguments
+}
+
+// GenerateGenesisAccountFromFile generates an account that is going to be directly
+// inserted into state
+func GenerateGenesisAccountFromFile(
+	filepath string,
+	constructorArgs []interface{},
+) (*chain.GenesisAccount, error) {
+	// Create the artifact from JSON
+	artifact, err := generateContractArtifact(filepath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Generate the contract ABI object
+	contractABI, err := abi.NewABI(string(artifact.ABI))
+	if err != nil {
+		return nil, fmt.Errorf("unable to create contract ABI, %w", err)
+	}
+
+	// Encode the constructor params
+	constructor, err := abi.Encode(
+		// Constructor arguments are passed in as an array of values.
+		// Structs are treated as sub-arrays with their corresponding values laid out
+		// in ABI encoding
+		normalizeConstructorArguments(constructorArgs),
+		contractABI.Constructor.Inputs,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("unable to encode constructor arguments, %w", err)
+	}
+
+	finalBytecode := append(artifact.Bytecode, constructor...)
+
+	// Create an instance of the state
+	st := itrie.NewState(itrie.NewMemoryStorage())
+
+	// Create a snapshot
+	snapshot := st.NewSnapshot()
+
+	// Create a radix
+	radix := state.NewTxn(st, snapshot)
+
+	// Create the contract object for the EVM
+	contract := runtime.NewContractCreation(
+		1,
+		types.ZeroAddress,
+		types.ZeroAddress,
+		staking.AddrStakingContract,
+		big.NewInt(0),
+		math.MaxInt64,
+		finalBytecode,
+	)
+
+	// Enable all forks
+	config := chain.ForksInTime{
+		Homestead:      true,
+		Byzantium:      true,
+		Constantinople: true,
+		Petersburg:     true,
+		Istanbul:       true,
+		EIP150:         true,
+		EIP158:         true,
+		EIP155:         true,
+	}
+
+	// Create a transition
+	transition := state.NewTransition(config, radix)
+
+	// Run the transition through the EVM
+	res := evm.NewEVM().Run(contract, transition, &config)
+	if res.Err != nil {
+		return nil, fmt.Errorf("EVM predeployment failed, %w", res.Err)
+	}
+
+	// After the execution finishes,
+	// the state needs to be walked to collect all touched
+	// storage slots
+	storageMap := make(map[types.Hash]types.Hash)
+
+	radix.GetRadix().Root().Walk(func(k []byte, v interface{}) bool {
+		if types.BytesToAddress(k) != staking.AddrStakingContract {
+			// Ignore all addresses that are not the one the predeployment
+			// is meant to run for
+			return false
+		}
```

Comment:
Can we define a helper function to walk and returns storageMap for specified address?

Reply:
Extracted:

[3b85428](https://github.com/0xPolygon/polygon-edge/pull/536/commits/3b854287ab6f1b526289f7acf39a5169a3f59e33)